### PR TITLE
[418504] Override Tester: Check for Synchronized

### PR DESCRIPTION
**Pull request by cdietrich**
[418504] Override Tester: Check for Synchronized

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=418504

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java [hunk 1]
```diff
@@ -1045,6 +1045,9 @@ protected void doCheckFunctionOverrides(EObject sourceElement, IResolvedOperatio
 				} else if (details.contains(OverrideCheckDetails.RETURN_MISMATCH)) {
 					error("The return type is incompatible with " + inherited.getSimpleSignature(), sourceElement,
 							returnTypeFeature(sourceElement), INCOMPATIBLE_RETURN_TYPE);
+				} else if (details.contains(OverrideCheckDetails.SYNCHRONIZED_MISMATCH)) {
+					warning("The overriden method is sychronized, the current one is not synchronized.", sourceElement,
```
Comment (kthoms) [inline org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java]: Typos: overriden -> overridden, sychronized -> synchronized

File: org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend [hunk 2]
```diff
@@ -75,6 +75,40 @@ class QuickfixTest extends AbstractXtendUITestCase {
...
 
 	@Test
+	def void missingSynchronized() {
+		setSeverity(MISSING_SYNCHRONIZED, "warning")		
+		create('XXX.xtend', '''
+			class Bar {
+				def synchronized int doSth() {
+					0
+				}
+
```
Comment (kthoms) [inline org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend]: remove obsolete empty lines, also below

Comment (cdietrich) [inline org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/QuickfixTest.xtend]: Can't the test framework wants them

**Pull request merged**